### PR TITLE
Feature/gh pages auto

### PR DESCRIPTION
**Pull request by Lypsolon**
This is an example of how we could tackle Repo-specific documentation without having to get everything to work under Docusaurus and also without needing to have a substantial developer docs section under the Ayon docs. 

PS: It was directly stolen from @robin-ynput on Discord. It's a great idea in my eyes. 

Also, the Doxygen dox is very limited right now as they only do code documentation, but that can be extended if we go this path.

## Changelog Description
Added GH workflow to allow for Github Pages to display Doxygen Documentation

## Additional info
added info to readme where to find GH Pages docs. 
The URI is currently guessed tho. 

Currently, the workflow targets Development in later iterations. I would like to get versioned docs inside of Doxygen [maybe with this](https://github.com/root-project/doxyvers) and target main and only build docs for: main+tag for version + develop

## Testing notes:
I'm not sure if testing is possible without pushing to develop. 

The only test I currently have is that I built a repo and copied and pasted the Docs folder from this repo into it. 
https://github.com/Lypsolon/PageWithCi

### CODE REVIEW

File: .github/workflows/static.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Simple workflow for deploying static content to GitHub Pages
+name: Deploy static content to Pages
+
+on:
+  # Runs on pushes targeting the default branch
+  push:
+    branches: ["devlop"]
```
Comment (BigRoy) [inline .github/workflows/static.yml:7]: Typo?
```suggestion
    branches: ["develop"]
```

File: README.md [hunk 2]
```diff
@@ -57,6 +57,11 @@ AYON compatible entity URIs through the
     git submodule update --init --recursive
     ```
 
+## DoxyGen Dox
+
+Can be found on the connected GH Page
+`https://ynput.github.io/ayon-usd-resolver/`
```
[the remark this answers is not in the document]
Comment (Lypsolon) [reply README.md]: fixed

**Comment by Lypsolon** [comment]
@dee-ynput based on the talk today and the fact that all the other Cpp tools already have there docs hosted can we approve this PR ? 

or at least approve it to the point where @BigRoy or @antirotor can check if they like the docs and then approve the merge ? 

i would like to get this baseline done so that all the tools get unified.

**Comment by Lypsolon** [comment]
PS: the ayon-cpp-api pointer is inline with Pinning file support branch. 
we need to see how we tell GH that we actually want a new version of the cpp api.

**Comment by Lypsolon** [comment]
im moving comments from the Pinning branch over to here.

**Comment by Lypsolon** [comment]
https://github.com/ynput/ayon-usd-resolver/pull/24/files/91486b85a93bc5c23027b0d60e1a48c5cf7f4f05#r1723056733
Roy I don't understand this :)

Dev Docs line 128 
This should always be the path used to open the stage Otherwise the pinning file
might not have the correct AssetIdentifier stored.\
aka: if this key points to a local path and you use an URI that points to the
local path the resolver wont be able to resolve.

**Comment by Lypsolon** [comment]
https://github.com/ynput/ayon-usd-resolver/pull/24/files/91486b85a93bc5c23027b0d60e1a48c5cf7f4f05#r1723056027
Typo sceene?

Also, why do we define the scene file in the JSON?
Dev Docs line 118 

```py
stage = Usd.Stage.Open(pinning_file_json["ayon_pinning_data_entry_sceene"])

**Comment by Lypsolon** [comment]
https://github.com/ynput/ayon-usd-resolver/pull/24/files/91486b85a93bc5c23027b0d60e1a48c5cf7f4f05#r1723052887
Could you explain more on this? Not sure why or why not?
Dev Docs line 81

**How to use it ?**\
There are 3 Env variables that control the Pinning file support

`ENABLE_STATIC_GLOBAL_CACHE`\
Enables or Disables static Cache creation. This effectively allows you to enable
or disable Pinning file support.\
as you might want to have a pinning file in your env vars but for e.g debugging
you don't want to activate the feature.

**Comment by Lypsolon** [comment]
https://github.com/ynput/ayon-usd-resolver/pull/24/files/91486b85a93bc5c23027b0d60e1a48c5cf7f4f05#r1715237381
Robin 
Personal thoughts, but I feel this documentation might be missing a quick example that put together all of the steps you detail.
E.g.

Set the Ayon resolver
Enable pinning feature support
Manipulate the cache (generate, add/remove entries)
Disable the caching

**Comment by Lypsolon** [comment]
https://github.com/ynput/ayon-usd-resolver/pull/24/files/8eaf6344058f864b3ec911a92c6accc47293d6ec#r1715247164
Robin
picky, as you wrote above, I think the term URI should be added as very meaningful in this section.

**Pull request merged**